pkg/vcs/vcs.go

```diff
@@ -58,3 +87,86 @@ func VCS(path string) (result v1.VCS) {
 
 	return
 }
+
+func ImageInfoFromApp(ctx context.Context, app *apiv1.App, cloneDir string) (string, string, error) {
+	vcs := app.Status.AppImage.VCS
+	if len(vcs.Remotes) == 0 {
+		return "", "", fmt.Errorf("clone can only be done on an app built from a git repository")
+	}
+	if vcs.Acornfile == "" || vcs.BuildContext == "" {
+		return "", "", fmt.Errorf("app is missing required vcs information, image must be rebuilt with a newer acorn cli")
+	}
+
+	// Create auth object to use when fetching and cloning git repos
+	auth, err := ssh.NewSSHAgentAuth("git")
+	if err != nil {
+		return "", "", err
+	}
+
+	for _, remote := range vcs.Remotes {
+		// Since we use ssh auth to clone the repo we need a git url but will sometimes get http urls
+		var gitUrl string
+		httpUrl, err := url.Parse(remote)
+		if err == nil {
+			gitUrl = fmt.Sprintf("git@%s:%s", httpUrl.Host, httpUrl.Path[1:])
+		} else {
+			gitUrl = remote
+		}
+
+		// Determine the repository name from the repo url
+		idx := strings.LastIndex(remote, "/")
+		if idx < 0 || idx >= len(remote)-1 {
+			fmt.Printf("failed to determine repository name %q\n", remote)
+			continue
+		}
+		workdir := filepath.Join(cloneDir, strings.TrimSuffix(remote[idx+1:], ".git"))
+
+		// Clone git repo and checkout revision
+		fmt.Printf("# Cloning repository %q into directory %q\n", gitUrl, workdir)
```

Comment:
nit: I don't think we start any other CLI log messages with `#` so it seems a bit odd to do it here.